### PR TITLE
Jaspreet

### PR DESCRIPTION
**Pull request by jaspreetkaursohal**
Creates the UI for how a post is going to be displayed.
Currently only implemented for text based posts and comments.
Next steps would involve pictures and markdown.

Created a new form for comments' content

Implemented logic and UI for:
1) When user likes/unlikes a post
2) Comments on a specific post
3) Viewing a post

To do either action, a user must be logged in.

I've tried implementing view tests for liking and viewing a post - critical feedback is highly requested and appreciated c:

### CODE REVIEW

File: quickcomm/views.py [hunk 1]
```diff
@@ -61,6 +62,55 @@ def login(request):
...
+
+    return render(request, "quickcomm/post.html", context)
+
+@login_required
+def post_like(request, post_id):
+    objects = Post.objects.all()
+    post = objects.get(pk=post_id)
+
+    if request.user.is_authenticated:
+        print("comes in here?")
```
Comment (micheal-chow) [inline quickcomm/views.py:85]: unnecessary print statement

File: quickcomm/urls.py [hunk 2]
```diff
@@ -8,4 +8,7 @@
     path('register/', views.register, name='register'),
     path('create/plain', views.create_post, name='create_plain'),
     path('create/markdown', views.create_markdown, name='create_markdown'),
+    path('post/<uuid:post_id>/', views.post_view, name='post_view'),
+    path('post/<uuid:post_id>/post_liked', views.post_like, name='post_like'),
+    path('post/<uuid:post_id>/post_comment', views.post_comment, name='post_comment')
```
Comment (micheal-chow) [inline quickcomm/urls.py:13]: this is probably fine for part 1, but once we need to join with other servers, we should follow the URLs in the spec. Which is to say we use posts/{POST_ID}/comments to post a comment (and the GET method on that same URL to view all comments on a given post).

File: quickcomm/tests/test_views.py [hunk 3]
```diff
@@ -68,3 +69,40 @@ def test_logout(self):
...
+        c.get('/post/'+str(author1.id)+"/",)
+        self.assertEqual(response.status_code, 302)
+    def test_like(self):
+        c = Client()
+        author1 = Author.objects.all()[0]
+        author2 = Author.objects.all()[1]
+        post = Post.objects.create(author=author1, title='My Post', source='http://someurl.ca', origin='http://someotherurl.ca', description='My Post Description', content_type='text/plain', content='My Post Content', visibility='PUBLIC', unlisted=False, categories='["test"]')
+        post.full_clean()
+
+        like = Like.objects.create(author=author2, post=post)
```
Comment (micheal-chow) [inline quickcomm/tests/test_views.py:106]: unused

File: quickcomm/tests/test_views.py [hunk 4]
```diff
@@ -68,3 +69,40 @@ def test_logout(self):
...
+        
+        self.author1 = Author.objects.create(user=self.user1, host='http://127.0.0.1:8000', display_name='user1', github='https://github.com/', profile_image='https://www.history.com/.image/c_fit%2Ccs_srgb%2Cfl_progressive%2Cq_auto:good%2Cw_620/MTU3ODc5MDg2NDM2NjU2NDU3/reagan_flags.jpg')
+        self.author1.save()
+
+        self.author2 = Author.objects.create(user=self.user2, host='http://127.0.0.1:8000', display_name='user2', github='https://github.com/', profile_image='https://www.history.com/.image/c_fit%2Ccs_srgb%2Cfl_progressive%2Cq_auto:good%2Cw_620/MTU3ODc5MDg2NDM2NjU2NDU3/reagan_flags.jpg')
+        self.author2.save()
+
+    def test_viewing_post(self):
+        author1 = Author.objects.all()[0]
+        author2 = Author.objects.all()[1]
```
Comment (micheal-chow) [inline quickcomm/tests/test_views.py:94]: unused

File: quickcomm/migrations/0001_initial.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
+# Generated by Django 4.1.7 on 2023-02-22 20:04
```
Comment (micheal-chow) [inline quickcomm/migrations/0001_initial.py:1]: This file should probably not be included.

**Review by micheal-chow — COMMENTED** [review]
Looks good except for the things I've addressed in other comments. There are a few things that will have to be added later but few changes required now as far as I can tell.

**Comment by Jdvdb** [comment]
The like feature isn't working for me. I am able to comment but clicking the like button refreshes the page and doesn't update the model. I tried it with 2 users but nothing is being updated in the database. Is the feature working for anyone else?

Other than that all tests are good, it looks great viewing a post and the comments are working well!

File: quickcomm/migrations/0001_initial.py (hunk 5 above)
Comment (rajanmaghera) [reply quickcomm/migrations/0001_initial.py:1]: I second that, let's not include migrations till the end.

File: quickcomm/tests/test_views.py [hunk 6]
```diff
@@ -68,3 +69,40 @@ def test_logout(self):
...
+        )
+
+        
+        self.author1 = Author.objects.create(user=self.user1, host='http://127.0.0.1:8000', display_name='user1', github='https://github.com/', profile_image='https://www.history.com/.image/c_fit%2Ccs_srgb%2Cfl_progressive%2Cq_auto:good%2Cw_620/MTU3ODc5MDg2NDM2NjU2NDU3/reagan_flags.jpg')
+        self.author1.save()
+
+        self.author2 = Author.objects.create(user=self.user2, host='http://127.0.0.1:8000', display_name='user2', github='https://github.com/', profile_image='https://www.history.com/.image/c_fit%2Ccs_srgb%2Cfl_progressive%2Cq_auto:good%2Cw_620/MTU3ODc5MDg2NDM2NjU2NDU3/reagan_flags.jpg')
+        self.author2.save()
+
+    def test_viewing_post(self):
```
Comment (rajanmaghera) [inline quickcomm/tests/test_views.py:92]: Try to include docstrings explaining what the test is doing.

File: quickcomm/urls.py (hunk 2 above)
Comment (rajanmaghera) [reply quickcomm/urls.py:13]: I actually spoke to @jaspreetkaursohal about this in person and okayed this, just for part 1. This URL is not meant to be used externally, only for our UI. In the future, the API will include a like, which will be client-side called via AJAX.

File: quickcomm/views.py [hunk 7]
```diff
@@ -61,6 +62,55 @@ def login(request):
...
+            author = Author.objects.all().get(user=request.user)
+            comment = Comment()
+            comment.post = post
+            comment.author = author
+            comment.comment = form.cleaned_data['comment']
+            comment.save()
+            return HttpResponseRedirect(reverse('post_comment', args=[post.pk]))
+        
+    else:
+        form = CreateCommentForm()
```
Comment (rajanmaghera) [inline quickcomm/views.py:111]: Remove these two lines, as they do not do anything. This new form doesn't go anywhere. 

Also the way the code is written, there will be no response if there is an error, but it's okay for now.

**Pull request merged**